src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -80,9 +80,15 @@ public override void OnStartActivity(Activity activity, object payload)
             var request = context.Request;
             if (!this.hostingSupportsW3C || !(this.options.Propagator is TextMapPropagator))
             {
+                var parentContext = new ActivityContext(
```

Comment:
Might be a bit faster if you promote activity.Context to a local. Eg:
```csharp
                var activityContext = activity.Context;
                var parentContext = new ActivityContext(
                    activityContext.TraceId,
                    activity.ParentSpanId,
                    activityContext.TraceFlags,
                    activityContext.TraceState,
                    true);
```
Not sure if the optimizer is smart enough to do that automatically.